src/jltypes.c

```diff
@@ -582,46 +582,102 @@ JL_DLLEXPORT jl_value_t *jl_type_union(jl_value_t **ts, size_t n)
     return tu;
 }
 
+// note: this is turned off as `Union` doesn't do such normalization.
+// static int simple_subtype(jl_value_t *a, jl_value_t *b)
+// {
+//     if (jl_is_kind(b) && jl_is_type_type(a) && jl_typeof(jl_tparam0(a)) == b)
+//         return 1;
+//     if (jl_is_typevar(b) && obviously_egal(a, ((jl_tvar_t*)b)->lb))
+//         return 1;
+//     return 0;
+// }
```

Comment:
I believe we should revisit this after #49279.

The first branch seems broken though, as
```julia
julia> T{A,B} = Union{Union, Type{Union{A,B}}}
Union{Type{Union{A, B}}, Union} where {A, B}

julia> T{Int,Int8}
Union

julia> T{Int,Union{}}
Union{Type{Int64}, Union}
```